### ISSUE TITLE
Breaks When Two Matrix Components are on an Entry 

### ISSUE THREAD

**Issue by DynamiteGoesBoom** [verbb/smith#34]
### Describe the bug

We have a Hero Builder and a Content Builder on our Entries like so:

![⚙-Team-Section-Gets-Some-Love-Get-Fido](https://github.com/verbb/smith/assets/39201467/96969db7-ad27-4a91-859b-ebc47f8bc135)

But now when we click on either type in the Hero Builder, nothin happens

Also the Paste doesn't really work when copying from the same Content Builder component on another page. 

### Steps to reproduce

1. add 2 matrix fields with multiple blocks 
2. copy block to new page
3. try adding a new block manually to the non-copied block

### Craft CMS version

Craft Pro 4.5.11.1

### Plugin version

2.0.0

**Comment by engram-design** [comment]
What's that UI above the Matrix block buttons? Another plugin, custom field, or something? It's not something I've seen before.

EDIT: Ah, this seems like it's the [Batch Actions](https://plugins.craftcms.com/batch-actions?craft4) plugin. Is this a Smith issue?

**Comment by DynamiteGoesBoom** [comment]
That would be Batch Actions. I use that along with Matrix Mate to add some nice AX touches. Smith looks like it integrates pretty well with it.

**Comment by engram-design** [comment]
Ah, this seems like its the [Batch Actions](https://plugins.craftcms.com/batch-actions?craft4) plugin. Is this a Smith issue?

**Comment by DynamiteGoesBoom** [comment]
Yep, you got it. But yes Smith is breaking the hero builder field. Once I turned it off the hero Builder started working again.

**Comment by DynamiteGoesBoom** [comment]
It's not a deal breaker as word on the street is that Smith-like functionality is coming to Craft 5. So if it's too big of a lift to get everything about Smith working, then I can wait too. I just like the idea of being able to easily clone matrix data. But not sure if the reward is worth the effort.

Thanks Josh!

**Comment by engram-design** [comment]
Hmm, so copy-paste across entries seems to be working okay for me, but then I probably need to test in combination with Matrix Mate and Batch Actions, in case there's some conflict there.

But yes, it's probably going to be largely no longer required in Craft 5, but waiting for the dev preview to settle on that one before making a final decision. Also seems like Batch Actions might do a better job anyway!

**Comment by DynamiteGoesBoom** [comment]
Yeah but Batch Actions doesn't copy/paste. i did find it interesting that BA and Smith played nicely together though. Although the functionality didn't work. But BA placed the Smith commands in as part of BA. If that makes sense. I didn't test Smith out by itself as BA does a bit more of what our clients like.

That sounds good about C5 and hopefully that all will be decided soon. Thanks for looking into this as i know you're quite the busy guy these days 🎉

**Comment by engram-design** [comment]
Oh, I thought it did, based on the screenshot above, but I see what Batch Actions is doing - neat!

**Comment by DynamiteGoesBoom** [comment]
Haha, no worries. It is a pretty nifty little plugin. If you had time to sort out the issue I'd love to see it. No biggie either way. Thanks again for all that you do!
